### PR TITLE
Improve encapsulation on lifecycle, ownership and payments

### PR DESCRIPTION
**Pull request by elopio**
This is part of #1174 
Requires #1254.

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

**Comment by frangio** [comment]
Rebased on `master`.

File: contracts/ownership/Ownable.sol [hunk 1]
```diff
@@ -22,14 +22,21 @@ contract Ownable {
...
    */
   constructor() public {
-    owner = msg.sender;
+    owner_ = msg.sender;
+  }
+
+  /**
+   * @return the address of the owner.
+   */
+  function getOwner() public view returns(address) {
```
Comment (frangio) [inline contracts/ownership/Ownable.sol]: I'm not fond of starting this convention. I feel like it goes against the rest of the ecosystem.

Comment (elopio) [reply contracts/ownership/Ownable.sol]: This is what we discussed about having verbs on function names. I won't like it, but I can remove the `get`s and try to convince the ecosystem to use them before 3.0.

Comment (frangio) [reply contracts/ownership/Ownable.sol]: Yes, I'd prefer to remove the `get`.

File: contracts/payment/RefundEscrow.sol [hunk 2]
```diff
@@ -16,25 +16,39 @@ contract RefundEscrow is Ownable, ConditionalEscrow {
...
-    state = State.Active;
+    beneficiary_ = _beneficiary;
+    state_ = State.Active;
+  }
+
+  /**
+   * @return the current state of the escrow.
+   */
+  function getState() public view returns(State) {
+    return state_;
```
[the remark this answers is not in the document]
Comment (frangio) [reply contracts/payment/RefundEscrow.sol]: I think the error just means that they're not the same thing. Try using `should.be.bignumber.equal` to get a nicer error message.

[the remark this answers is not in the document]
Comment (elopio) [reply contracts/payment/RefundEscrow.sol]: that helped, thanks!

File: contracts/ownership/Claimable.sol [hunk 3]
```diff
@@ -10,30 +10,35 @@ import "./Ownable.sol";
...
+    pendingOwner_ = newOwner;
   }
 
   /**
    * @dev Allows the pendingOwner address to finalize the transfer.
    */
   function claimOwnership() public onlyPendingOwner {
-    emit OwnershipTransferred(owner, pendingOwner);
-    owner = pendingOwner;
-    pendingOwner = address(0);
```
Comment (frangio) [inline contracts/ownership/Claimable.sol]: This line should stay because it's not part of `_transferOwnership`. But we've removed `Claimable` in #1274.

Comment (frangio) [reply contracts/ownership/Claimable.sol]: Ah, #1274 is not merged yet.

Comment (elopio) [reply contracts/ownership/Claimable.sol]: indeed, and no test failed :-1:

File: contracts/ownership/Claimable.sol [hunk 4]
```diff
@@ -40,5 +40,6 @@ contract Claimable is Ownable {
    */
   function claimOwnership() public onlyPendingOwner {
     _transferOwnership(pendingOwner_);
+    pendingOwner = address(0);
```
[the remark this answers is not in the document]
Comment (elopio) [reply contracts/ownership/Claimable.sol]: goddammit! I should have breakfast first :D

**Comment by elopio** [comment]
@frangio resolved

**Comment by elopio** [comment]
There's something weird with the coverage report. The lines it report uncovered have specific tests added on this PR, and that lead to full coverage on previous commits. Maybe, I broke something on one merge, or maybe it's coveralls being nuts.

**Pull request merged**